changelogs/unreleased/gh-9020-fix-use-after-free-on-fiber-finish.md

```diff
@@ -0,0 +1,4 @@
+## bugfix/core
+
+* Fixed a bug that on fiber finish in rare occasion it can write to a memory
+  owned by another thread (gh-9020).
```

Comment:
Let's just say that this patch fixes a use-after-free bug in the fiber recycling code.

Reply:
```md
## bugfix/core

* Fixed a use-after-free bug in the fiber recycling code (gh-9020).
```